src/Groups/GAPGroups.jl

```diff
@@ -1741,31 +1741,34 @@ false
 
 
 @doc raw"""
-    is_full_fp_group(G::FPGroup)
+    full_group(G::T) where T <: Union{SubFPGroup, SubPcGroup}
+    full_group(G::T) where T <: Union{FPGroup, PcGroup}
 
-Return `true` if `G` has been constructed as a free group or
-a quotient of a free group, and `false` otherwise.
-
-Note that also subgroups of groups of type `FPGroup` have the type `FPGroup`,
-and functions such as [`relators`](@ref) do not make sense for proper
-subgroups.
+Return `F, emb` where `F` is the full pc group of f.p. group of which `G`
+is a subgroup, and `emb` is an embedding of `G` into `F`.
 
 # Examples
 ```jldoctest
-julia> f = free_group(2);  is_full_fp_group(f)
-true
+julia> G = perfect_group(FPGroup, 60, 1);
 
-julia> s = sub(f, gens(f))[1];  is_full_fp_group(s)
-false
+julia> H = sylow_subgroup(G, 2)[1];
 
-julia> q = quo(f, [gen(f,1)^2])[1];  is_full_fp_group(q)
+julia> full_group(H)[1] == G
 true
 
-julia> u = sub(q, gens(q))[1];  is_full_fp_group(u)
-false
+julia> full_group(G)[1] == G
+true
 ```
 """
-is_full_fp_group(G::FPGroup) = GAPWrap.IsFpGroup(GapObj(G))
+function full_group(G::T) where T <: Union{SubFPGroup, SubPcGroup}
```

Comment:
```suggestion
function full_group(G::Union{SubFPGroup, SubPcGroup})
```
If you don't use `T` in the body, it is more efficient (in terms of compile time) to not have the type parameter at all